### PR TITLE
Electbackup

### PR DESCRIPTION
**Pull request by marcosrdz**
support importing electrum wallet back up files. 

support for importing seed/xpub from backup file

open your electrum wallet, click file> save backup
Screen Shot 2020-09-30 at 2 26 11 PM

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/jc9gntgz9tzcexbpnqm5y4zzfr

[download apk](http://filestorage.bluewallet.io:1488/1601431855.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/etatcndt52thhwa0nqt890mng8

[download apk](http://filestorage.bluewallet.io:1488/1601435386.apk)

File: class/wallets/abstract-wallet.js [hunk 1]
```diff
@@ -165,7 +165,10 @@ export class AbstractWallet {
 
     try {
       const parsedSecret = JSON.parse(this.secret);
-      if (parsedSecret && parsedSecret.keystore && parsedSecret.keystore.xpub) {
+      if (parsedSecret && parsedSecret.keystore && parsedSecret.keystore.secret) {
+        // It is an Electrum Wallet Backup
+        this.secret = parsedSecret.keystore.seed;
+      } else if (parsedSecret && parsedSecret.keystore && parsedSecret.keystore.xpub) {
```
Comment (limpbrains) [inline class/wallets/abstract-wallet.js:171]: `?.` would be nice

File: class/deeplink-schema-match.js [hunk 2]
```diff
@@ -208,6 +226,13 @@ class DeeplinkSchemaMatch {
     );
   }
 
+  static isElectrumWalletBackupFile(filePath) {
+    return (
+      (filePath.toLowerCase().startsWith('file:') || filePath.toLowerCase().startsWith('content:')) &&
+      filePath.toLowerCase().endsWith('.backup')
+    );
+  }
```
Comment (limpbrains) [inline class/deeplink-schema-match.js:234]: Please write a test for it in `deeplink-schema-match.test.js`

**Review by limpbrains — CHANGES_REQUESTED** [review]

**Comment by limpbrains** [comment]
@marcosrdz can you please write a few words of what this PR does. As I understand it is possible to create electrum backup is `.psbt` and extract seed from it, am I right?

**Comment by marcosrdz** [comment]
> @marcosrdz can you please write a few words of what this PR does. As I understand it is possible to create electrum backup is `.psbt` and extract seed from it, am I right?

updated description

**Review by limpbrains — CHANGES_REQUESTED** [review]
Please tests at least

File: ios/Podfile.lock [hunk 3]
```diff
@@ -724,4 +724,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: e9c5efd531ca5ac67a4b743a179eeefb322cf387
 
-COCOAPODS: 1.10.0.beta.2
+COCOAPODS: 1.9.3
```
Comment (Overtorment) [inline ios/Podfile.lock:727]: downgrade. is it ok?

File: screen/send/ScanQRCode.js [hunk 4]
```diff
@@ -134,15 +134,22 @@ const ScanQRCode = () => {
...
+            : [DocumentPicker.types.allFiles],
+      });
       const file = await RNFS.readFile(res.uri);
       const fileParsed = JSON.parse(file);
-      if (fileParsed.keystore.xpub) {
+      if (fileParsed.keystore.seed) {
+        onBarCodeRead({ data: fileParsed.keystore.seed, additionalProperties: { label: file.name } });
+      } else if (fileParsed.keystore.xpub) {
         let masterFingerprint;
-        if (fileParsed.keystore.ckcc_xfp) {
```
Comment (Overtorment) [inline screen/send/ScanQRCode.js:142]: i think I refactored all that mess in master. 
long story short, it doesn't belong here. import code is grouped in other place

File: class/wallets/abstract-wallet.js [hunk 5]
```diff
@@ -165,7 +165,10 @@ export class AbstractWallet {
 
     try {
       const parsedSecret = JSON.parse(this.secret);
-      if (parsedSecret && parsedSecret.keystore && parsedSecret.keystore.xpub) {
+      if (parsedSecret && parsedSecret.keystore && parsedSecret.keystore.secret) {
```
Comment (Overtorment) [inline class/wallets/abstract-wallet.js:168]: tbh i dont remember why we have some import code in abstractwallet class.
anyway, it is mostly watch-only stuff (like, fancy wallet descriptor format `[8cce63f8/84h/0h/0h]zpub6s2RJ9qAEBW8Abhojs6LyDzF7gttcDr6EsR3Umu2aptZBb45e734rGtt4KqsCMmNyR1EEzUU2ugdVYez2VywQvAbBjUSKn8ho4Zk2c5otkk` )

not sure electrum backup stuff belongs here.

File: screen/send/ScanQRCode.js (hunk 4 above)
Comment (Overtorment) [reply screen/send/ScanQRCode.js:142]: `ScanQRCode.js` is only for scaning qr / opening file and returning its contents

File: class/deeplink-schema-match.js (hunk 2 above)
Comment (Overtorment) [reply class/deeplink-schema-match.js:234]: +1. tests are needed. we now have `tests/unit/fixtures/` where you can example backup file if you need

**Review by Overtorment — CHANGES_REQUESTED** [review]

**Comment by ncoelho** [comment]
This one is super useful @marcosrdz

**Comment by marcosrdz** [comment]
> This one is super useful @marcosrdz

@Overtorment  didnt the multisig work include being able to import electrum backup files?

**Comment by Overtorment** [comment]
no, import code is inside multisig wallet class, so if its single sig it wont impor tit

**Pull request closed without merging**